### DISCUSSION TITLE
无法连续点击

### DISCUSSION THREAD

**Discussion by haipi12138** [gkd-kit/gkd#617] [Q&A] [answered]
![屏幕截图 2024-05-11 212449](https://github.com/gkd-kit/gkd/assets/103874387/4c619e51-18db-4eab-a108-006036c7be4a)
这是一个被破解的漫画软件点击广告会直接跳过继续阅读 但每次点进来都会弹出需要点击“确定” 尝试自己写了一个规则但只会生效一次 或者只会在短时间内弹出窗口后才会点击（即便是弹出了四五次，具体为回到书架 在进入漫画就会弹窗） 一旦2s左右内没有弹窗之后规则就像失效了 除非切回桌面在切回软件回到刚开始的状态 以下是我的规则：

```json5
{
  name: '确定',
     key: 1,
     matchDelay: 7000,
     actionCd: 100,
     snapshotUrls: [
    'https://i.gkd.li/snapshot/1715330117510',
    ''
  ],
  activityIds: [
    'top.guuguo.read_info.MainActivity'
  ],
  rules: [
    {
      matches: [
        '[desc="确定"]'
      ]
    }
  ]
}
```

**Comment by lisonge** [comment]
快照链接是错误的

**Reply by haipi12138** [reply]
![1](https://github.com/gkd-kit/gkd/assets/103874387/e7b3676d-b6bb-4c0c-aa1c-d7cdb1626a95)
![2](https://github.com/gkd-kit/gkd/assets/103874387/76d9fd6f-0388-463b-8900-44ef2b0e1cc7)
没问题啊 大佬
actionCd: 100,
matchDelay: 7000,
这俩也是抄的你的一个问题里的类似情况

**Reply by lisonge** [reply]
![image](https://github.com/gkd-kit/gkd/assets/38517192/bf6759b4-e774-4e98-9d97-dbf5b9acd450)

**Reply by lisonge** [reply]
可以设置一个较长的 <https://gkd.li/api/interfaces/RawAppRule.html#forcedtime>

**Reply by haipi12138** [reply]
https://i.gkd.li/snapshot/1715330117510
https://i.gkd.li/i/15272567
这俩种快照链接都试过 感觉区别不大 forcedTime也试过设置了10000 也没啥效果

**Reply by lisonge** [reply]
你换个新的浏览器分别打开 https://i.gkd.li/snapshot/1715330117510 和 https://i.gkd.li/i/15272567 就知道它们的区别了

**Reply by haipi12138** [reply]
对 其中一个更快 但是我在规则里分别用过俩个链接 都区别不大 也是都只能打开一次

**Reply by lisonge** [reply, marked as answer]
forcedTime: 10000 也才 10s，太短了，你改成 forcedTime:120000 呢

**Reply by lisonge** [reply]
> 对 其中一个更快 但是我在规则里分别用过俩个链接 都区别不大 也是都只能打开一次

我的意思是，你用一个新安装的其它浏览器，或者其它设备的浏览器打开这两种链接就知道区别了

**Reply by haipi12138** [reply]
卧槽设置120s现在就可以见到弹窗就秒杀了 大佬牛皮 ，还有换了ios试了下长快照链接会显示快照丢失啥的，但不知道为啥之前用chrome 换了edge 俩链接就都可打开 可能是edge又同步了吧 虽然我记得把它关了 总之大佬牛批

**Reply by lisonge** [reply]
我发现 如果你先打开 <https://i.gkd.li/i/15272567> 那就看不出这两个链接的区别，因为它们的区别是

<https://i.gkd.li/snapshot/1715330117510> 是本地链接，只能在浏览器已经访问/存储过这个快照之后才能正常访问

而 <https://i.gkd.li/i/15272567> 是远程链接，快照文件已经上传到云端，可以在任意浏览器打开访问

你刚刚没有看出区别，我猜测是先打开了 <https://i.gkd.li/i/15272567> 导致浏览器里面有存储记录从而导致后续的 <https://i.gkd.li/snapshot/1715330117510> 也能正常打开

所以根据上面说的原理，如果你在一个新的浏览器(也就是我这边的浏览器)去打开 <https://i.gkd.li/snapshot/1715330117510> 是无法访问的

**Reply by haipi12138** [reply]
确实应该就是这样 看了下历史记录edge先访问了云端链接 又访问了本地链接

**Reply by haipi12138** [reply]
卧槽太蠢了我，forcedTime是一段时间内，主动查询。 我之前看成了 一段时间后才主动查询 哎~看来这个Q本来不必存在 太蠢了太蠢了
